internal/supervisor/import.go

```diff
@@ -0,0 +1,408 @@
+// from ready Imports produced by the AWAT. It periodically queries
+// the AWAT for Imports waiting to be performed and then performs
+// imports serially
+type ImportSupervisor struct {
+	awsClient   toolsAWS.AWS
+	awatClient  AWATClient
+	logger      logrus.FieldLogger
+	store       installationStore
+	provisioner importProvisioner
+	ID          string
+}
+
+type importProvisioner interface {
+	ExecClusterInstallationCLI(cluster *model.Cluster, clusterInstallation *model.ClusterInstallation, args ...string) ([]byte, error)
+}
+
+type mmctl struct {
+	cluster             *model.Cluster
+	clusterInstallation *model.ClusterInstallation
+	installation        *model.Installation
+
+	*ImportSupervisor
+}
+
+func (m *mmctl) Run(args ...string) ([]byte, error) {
+	args = append([]string{"mmctl", "--format", "json", "--local"}, args...)
+	return m.provisioner.ExecClusterInstallationCLI(m.cluster, m.clusterInstallation, args...)
+}
+
+type team struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type jobResponse struct {
+	ID       string `json:"id"`
+	Status   string `json:"status"`
+	Progress int    `json:"progress"`
+
+	jobResponseData `json:"data"`
+}
+
+type jobResponseData struct {
+	Error      string `json:"error"`
+	LineNumber int    `json:"line_number"`
+	ImportFile string `json:"import_file"`
+}
+
+// NewImportSupervisor creates a new Import Supervisor
+func NewImportSupervisor(awsClient toolsAWS.AWS, awat AWATClient, store installationStore, provisioner importProvisioner, logger logrus.FieldLogger) *ImportSupervisor {
+	return &ImportSupervisor{
+		awsClient:   awsClient,
+		awatClient:  awat,
+		store:       store,
+		logger:      logger,
+		provisioner: provisioner,
+
+		// TODO replace this with the Pod ID from env var
+		ID: model.NewID(),
+	}
+}
+
+// Do checks to see if there is an Import that is ready to be
+// imported, and if so, does that. Otherwise, it does nothing.
+func (is *ImportSupervisor) Do() error {
+	work, err := is.awatClient.GetTranslationReadyToImport(
+		&awat.ImportWorkRequest{
+			ProvisionerID: is.ID,
+		})
+	if err != nil {
+		return errors.Wrap(err, "failed to get a ready Import from the AWAT")
+	}
+	if work == nil {
+		// nothing to do
+		return nil
+	}
+	defer func() {
+		err = is.awatClient.ReleaseLockOnImport(work.ID)
+		if err != nil {
+			is.logger.WithError(err).Warnf("failed to release lock on Import %s", work.ID)
+		}
+	}()
+
+	err = is.importTranslation(work)
+	if err != nil {
+		is.logger.WithError(err).Errorf("failed to perform work on Import %s", work.ID)
+	}
+
+	errorString := ""
+	if err != nil {
+		errorString = err.Error()
+	}
+
+	cerr := is.awatClient.CompleteImport(
+		&awat.ImportCompletedWorkRequest{
+			ID:         work.ID,
+			CompleteAt: time.Now().UnixNano() / int64(time.Millisecond),
+			Error:      errorString,
+		})
+	if cerr != nil {
+		is.logger.WithError(cerr).Errorf("failed to notify AWAT that Import %s is complete", work.ID)
```

Comment:
Is it okay if we just proceed on this?

Reply:
Yeah, we want to return the actual error, in `err`, up the call stack, after sending this

Reply:
```suggestion
	err = is.importTranslation(work)
	if err != nil {
		is.logger.WithError(err).Errorf("failed to perform work on Import %s", work.ID)
	}

	errorString := ""
	if err != nil {
		errorString = err.Error()
	}
	{
		err := is.awatClient.CompleteImport(
			&awat.ImportCompletedWorkRequest{
				ID:         work.ID,
				CompleteAt: time.Now().UnixNano() / int64(time.Millisecond),
				Error:      errorString,
			})
		if err != nil {
			is.logger.WithError(cerr).Errorf("failed to notify AWAT that Import %s is complete", work.ID)
		}
	}
	return err
```

Reply:
Yes and no. It's okay from the perspective of the Provisioner, which is why this error is merely logged out and then the previous error, if it exists, is returned.

However, if this error occurs, the Import in the AWAT will never reach the completed state. Originally I considered this sub-optimal but OK for an MVP, however, after giving this more thought, I believe I've implemented this step while ignoring my design, by mistake ( :man_facepalming: ), and this final state transfer should probably be handled by the AWAT by polling the Provisioner periodically for the Installation to return to `stable` after an import is started. I think that would be more robust, overall.

I'm thinking that I should actually remove this stanza entirely from this PR and open a new PR against the AWAT with the correct functionality, rather than merge this as-is. Does that sound reasonable to you @gabrieljackson ?

With that said, I think after removing this stanza, this PR should still be OK to be merged since it will still include the entire change for the Provisioner.

Reply:
Yep, sounds good to me!